File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        for(int i = 0; i < flowerbed.size(); i++){
            bool isLeftEmpty = (i == 0 || flowerbed[i-1] == 0);
            bool isRightEmpty = (i == flowerbed.size() - 1 || flowerbed[i+1] == 0);
            if(isLeftEmpty && isRightEmpty && flowerbed[i] == 0){
                n--;
                flowerbed[i] = 1;
            }
        }
        return n <= 0;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        for i in range(0, len(flowerbed)):
            isLeftEmpty = (i == 0 or flowerbed[i-1] == 0)
            isRightEmpty = (i == len(flowerbed) - 1 or flowerbed[i+1] == 0)
            if(isLeftEmpty and isRightEmpty and flowerbed[i] == 0):
                flowerbed[i] = 1
                n -= 1
        return n <= 0
        